src/Features/Core/Portable/ConvertLinq/AbstractConvertLinqQueryToLinqMethodProvider.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.ConvertLinq
+{
+    internal abstract class AbstractConvertLinqQueryToLinqMethodProvider : AbstractConvertLinqProvider
+    {
+        protected abstract class Analyzer<TSource, TDestination> : AnalyzerBase<TSource, TDestination>
+            where TSource : SyntaxNode
+            where TDestination : SyntaxNode
+        {
+            public Analyzer(SemanticModel semanticModel, CancellationToken cancellationToken) : base(semanticModel, cancellationToken) { }
+
+            protected ImmutableArray<IAnonymousFunctionOperation> FindAnonymousFunctionsFromParentInvocationOperation(SyntaxNode node)
+            {
+                var operation = FindParentInvocationOperation(node);
+
+                if (operation != null)
+                {
+                    return ImmutableArray
+                        .CreateRange(operation.Arguments.Where(a => a.Value.Kind == OperationKind.DelegateCreation)
```

Comment:
>a.Value.Kind == OperationKind.DelegateCreation [](start = 68, length = 46)

I think we cannot assume that there is a delegate creation on top of the lambda. It could be a conversion to an expression tree. I recommend to relax this condition accordingly.